src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_MULTIPLIER = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[])new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Choose index less or equal than " + size
+                    + "of ArrayList<T>. This index incorrect " + index);
+        }
+        checkSize();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int additionalLength = list.size();
+        T[] listToArray = (T[]) new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            listToArray[i] = list.get(i);
+        }
+        System.arraycopy(listToArray, 0, elements, size, additionalLength);
+        size += additionalLength;
```

Comment:
I'd recommend iterating throughout the list and adding new elements using your add() method. There are a lot of issues with your current solution. Why do you call list.size() twice when you already have additionalLength variable? What will happen, if your elements array is too small to handle all additional elements?